pkg/metainfo/kvmetainfo/objects.go

```diff
@@ -307,3 +345,32 @@ func convertTime(ts *timestamp.Timestamp) time.Time {
 	}
 	return t
 }
+
+type mutableObject struct {
+	db   *DB
+	info storj.Object
+}
+
+func (object *mutableObject) Info(ctx context.Context) (storj.Object, error) {
```

Comment:
@egonelbre Can this method signature be simplified to:
```go
Info() storj.Object
```

Reply:
I've been wondering that myself. The problem is that the info may change while changing concurrently, so it would need a way to update itself. I mean it could have an extra method `UpdateInfo(ctx context.Context)` and then keep `Info` same as the readonly stream.

But sure, let's simplify for now.